### PR TITLE
add SupportsTls11 and SupportsTls12 to PlatformDetection avoid OS versions in tests

### LINKED ISSUE #39445: ServerAsyncAuthenticate_SniSetVersion_Success failing on Debian10
```
   System.Net.Security.Tests.ServerAsyncAuthenticateTest.ServerAsyncAuthenticate_SniSetVersion_Success(version: Tls11) [FAIL]
      System.AggregateException : One or more errors occurred. (One or more errors occurred. (Authentication failed, see inner exception.)) (One or more errors occurred. (Authentication failed, see inner exception.))
      ---- System.AggregateException : One or more errors occurred. (Authentication failed, see inner exception.)
      -------- System.Security.Authentication.AuthenticationException : Authentication failed, see inner exception.
      ------------ Interop+OpenSsl+SslException : SSL Handshake failed with OpenSSL error - SSL_ERROR_SSL.
      ---------------- Interop+Crypto+OpenSslCryptographicException : error:1409442E:SSL routines:ssl3_read_bytes:tlsv1 alert protocol version
      ---- System.AggregateException : One or more errors occurred. (Authentication failed, see inner exception.)
      -------- System.Security.Authentication.AuthenticationException : Authentication failed, see inner exception.
      ------------ Interop+OpenSsl+SslException : SSL Handshake failed with OpenSSL error - SSL_ERROR_SSL.
      ---------------- Interop+Crypto+OpenSslCryptographicException : error:14209102:SSL routines:tls_early_post_process_client_hello:unsupported protocol

```

It seems like Debian no longer allows TLS11 requested by the test.

### PR DESCRIPTION
**Pull request by wfurt**
It seems like Debian10 does not want to do TLS1.1 by default. We may see more of the in the future.

fixes #39445

### CODE REVIEW

**Comment by VSadov** [comment]
This will have to be ported to Preview8 branch as well since there are failures there too.

**Comment by danmosemsft** [comment]
@dotnet/ncl this is hitting a few jobs, could someone please review, and we can fix preview 8 also?

reopening PR to pick up fixes to WASM tests..

File: src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs [hunk 1]
```diff
@@ -99,7 +99,7 @@ public async Task ServerAsyncAuthenticate_SimpleSniOptions_Success()
             }
         }
 
-        [ConditionalTheory(nameof(IsNotWindows7))]
+        [ConditionalTheory(nameof(SupportsTls11))]
```
Comment (ManickaP) [inline src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs:102]: Does this mean that `SslProtocols.Tls12` won't be tested if the machine doesn't support `SupportsTls11`?
[`SupportsTls12`](https://github.com/dotnet/runtime/pull/39482/files#diff-03683e4fd9311d6aa15a7bdca89a95b2R130) works for `Debian10` if I get it right. Shouldn't we rather move the argument values in `MemberData` method and decide whether to test it per each `SslProtocols` value?

Comment (wfurt) [reply src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs]: I wanted to keep it simple. but I guess this is best option long run. I think I'll do more cleaning, keeping this small for now to stabilize CI.

File: src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs [hunk 2]
```diff
@@ -247,6 +244,18 @@ public static IEnumerable<object[]> ProtocolMismatchData()
...
         }
 
+        public static IEnumerable<Object[]> SupportedProtocolData()
+        {
+            if (PlatformDetection.SupportsTls11)
+            {
+                yield return new object[] { SslProtocols.Tls11 };
+            }
+
+            if (!PlatformDetection.SupportsTls12)
```
Comment (ManickaP) [inline src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs:254]: Is the `!` (not) correct here?

Comment (wfurt) [reply src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs]: fixed. I did some experiments and forget it to take it out again ;(

**Review by ManickaP — APPROVED** [review]
LGTM, thanks!

**Pull request merged**